docs/reference/mapping/types/parent-join.asciidoc

```diff
@@ -0,0 +1,363 @@
+[[mapping-parent-join-field]]
+=== `join` datatype
+
+The `join` datatype is a special field that creates
+parent/child relation within documents of the same index.
+This field defines a set of possible relations within the documents,
+each relation being a parent name and a child name.
+A single parent/child relation can be defined as follow:
+
+[source,js]
+--------------------------------------------------
+PUT my_index
+{
+  "mappings": {
+    "doc": {
+      "properties": {
+        "my_join_field": { <1>
+          "type": "join",
+          "my_parent": "my_child" <2>
+        }
+      }
+    }
+  }
+}
+--------------------------------------------------
```

Comment:
I 
 changed the examples in https://github.com/elastic/elasticsearch/pull/25227/commits/a088fcf0794eeb0432249472b2221350990e86c0 with a focus on single parent/child relation.